app/components/items/download-qr-code.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from "react";
+import { motion } from "framer-motion";
+import { atom, useAtom } from "jotai";
+import { tw } from "~/utils";
+import { CrossIcon } from "../icons";
+import { Button } from "../shared";
+
+const sidebarStatusAtom = atom(false);
+const toggleSidebarAtom = atom(
+  (get) => get(sidebarStatusAtom),
+  (get, set) => set(sidebarStatusAtom, !get(sidebarStatusAtom))
+);
+
+const DownloadQrCode = () => {
+  const [isSidebarOpen, toggleSidebar] = useAtom(toggleSidebarAtom);
+
+  useEffect(()=>{
+    if(isSidebarOpen) {
+      document.body.classList.add("overflow-hidden")
+    } else{
+      document.body.classList.remove("overflow-hidden")
+    }
+  }, [isSidebarOpen]);
+
+  return (
+    <div>
+      <Button icon="barcode" onClick={toggleSidebar} variant="secondary">
+        Download QR Tag
+      </Button>
+      <div className={tw("fixed right-0 top-0 z-10 h-screen w-screen bg-gray-25/70 backdrop-blur transition duration-300 ease-in-out", isSidebarOpen ? "visible" : "invisible opacity-0")}></div>
```

Comment:
Clicking on this should also toggle the sidebar, so if its open and the user clicks on the blury overlay, it should close it.

Reply:
okay